packages/kit/src/runtime/client/router.js

```diff
@@ -449,6 +450,10 @@ export class Router {
 		}
 
 		if (details) {
+			if (this.is_fallback && info.url.pathname !== location.pathname) {
```

Comment:
what is the second half of this check doing? I don't understand how an app can be initialized as a fallback/SPA and then later not become one. Maybe this variable is representing something else in this case as should have a different name?

Reply:
an SPA can have _some_ pages prerendered. the bug we're trying to avoid is infinite reloads when the initial hydration fails (or subsequent invalidations that aren't caused by navigation), rather than preventing full page reloads across the board